zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1844,6 +1849,61 @@ public void onParaInfosReceived(String noteId, String paragraphId,
     }
   }
 
+  @Override
+  public List<ParagraphInfo> getParagraphList(String user, String noteId)
+      throws TException, ServiceException {
+    try {
+      AuthenticationInfo authInfo = new AuthenticationInfo();
+      authInfo.setUser(user);
+
+      Set<String> userAndRoles = new HashSet<>();
+      userAndRoles.add(user);
+
+      ServiceContext serviceContext = new ServiceContext(authInfo, userAndRoles);
+      Note note = getNotebookService().getNote(noteId, serviceContext,
```

Comment:
NotebookService is usually for external call (such as websocket or rest api). Because only websocket & rest api needs callback. Here we can directly call notebook() to get Note and check permission.

Reply:
@zjffdu  Done. CI pass. https://travis-ci.org/liuxunorg/zeppelin/builds/501825407